api/envoy/extensions/transport_sockets/tls/v3/tls.proto

```diff
@@ -124,8 +126,25 @@ message DownstreamTlsContext {
   OcspStaplePolicy ocsp_staple_policy = 8 [(validate.rules).enum = {defined_only: true}];
 }
 
+// TLS key log configuration.
+// The key log file format is "format used by NSS for its SSLKEYLOGFILE debugging output" (text taken from openssl man page)
+message TlsKeyLog {
+  // The path to save the TLS key log.
+  string tls_keylog_path = 1;
+
+  // The src IP address that will be used to filter the connection which should save the TLS key log
+  config.core.v3.Address tls_keylog_src = 2;
```

Comment:
What happens if the src and dst addresses are not set?  In that case, does it log keys for all connections?  Please document.

Reply:
I think this should be a `repeated core.v3.CidrRange`, and be required to be non-empty (can add '0.0.0.0/0' if you want to match everything). There is already code to convert that to an `LcTrie` and match addresses on it.

Reply:
@markdroth,  if src is not set, only check dst. If dst is not set , only check src. If neither is set, it will log keys for all connections.